tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_PatternTests.cpp

```diff
@@ -0,0 +1,373 @@
+        PatternCallback pattern = ::testing::get<0>(GetParam());
+        CGFloat alpha = ::testing::get<1>(GetParam());
+        float xStep = ::testing::get<2>(GetParam());
+        float yStep = ::testing::get<3>(GetParam());
+        CGRect rect = ::testing::get<4>(GetParam());
+        char rectDesc[100];
+        snprintf(rectDesc, sizeof(rectDesc), "(%0.0f.%0.0f)%0.0fx%0.0f", rect.origin.x, rect.origin.y, rect.size.width, rect.size.height);
+        return CFStringCreateWithFormat(
+            nullptr,
+            nullptr,
+            CFSTR("TestImage.CGContext.CGPatternColoredRectBasedStroke.%s.rect.%s.xStep.%.02f.yStep.%.02f.alpha%.02f.png"),
+            pattern.name,
+            rectDesc,
+            xStep,
+            yStep,
+            alpha);
+    }
+};
+
+TEST_P(CGPatternColoredRectBasedStroke, PatternStrokeRegion) {
+    PatternCallback pattern = ::testing::get<0>(GetParam());
+    CGFloat alpha = ::testing::get<1>(GetParam());
+    float xStep = ::testing::get<2>(GetParam());
+    float yStep = ::testing::get<3>(GetParam());
+    CGRect rect = ::testing::get<4>(GetParam());
+
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    _SetPatternForStroke(context, rect, xStep, yStep, pattern.callback, alpha);
+    CGRect borderRect = CGRectInset(bounds, 30, 50);
+    CGContextSetLineWidth(context, 45);
+    CGContextStrokeRect(context, borderRect);
+}
+
+static CGRect tileRects[] = { CGRectMake(0, 0, 100, 100), CGRectNull };
+static float tileSteps[] = { 100, 50 };
+
+PatternCallback squaredPatternCallbacks[] = { { "drawStencilStar", &drawStencilStar },
+                                              { "drawPatternWindowsLogo", &drawPatternWindowsLogo } };
+INSTANTIATE_TEST_CASE_P(CGPatternTests,
+                        CGPatternColoredRectBasedStroke,
+                        ::testing::Combine(::testing::ValuesIn(squaredPatternCallbacks),
+                                           ::testing::ValuesIn(alphas),
+                                           ::testing::ValuesIn(tileSteps),
+                                           ::testing::ValuesIn(tileSteps),
+                                           ::testing::ValuesIn(tileRects)));
+
+//<PatternCallback,alpha,xstep,ystep,rect>
+class CGPatternColoredRectBasedFill
+    : public WhiteBackgroundTest<>,
+      public ::testing::WithParamInterface<::testing::tuple<PatternCallback, CGFloat, float, float, CGRect>> {
+    CFStringRef CreateOutputFilename() {
+        PatternCallback pattern = ::testing::get<0>(GetParam());
+        CGFloat alpha = ::testing::get<1>(GetParam());
+        float xStep = ::testing::get<2>(GetParam());
+        float yStep = ::testing::get<3>(GetParam());
+        CGRect rect = ::testing::get<4>(GetParam());
+        char rectDesc[100];
+        snprintf(rectDesc, sizeof(rectDesc), "(%0.0f.%0.0f)%0.0fx%0.0f", rect.origin.x, rect.origin.y, rect.size.width, rect.size.height);
```

Comment:
I think this is a case where the default name is better than a custom one.  Might start running into path length issues too...